### PR TITLE
[automate-3273] clean up IAM version references in teams-service

### PR DESCRIPTION
**Pull request by bcmdarroch**
### :nut_and_bolt: Description: What code changed, and why?

Just some cleanup. It's like the beta never happened.. 🔮 

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] Tests added/updated?
- [x] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/teams-service/README.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ make test_with_db
 If you wish to run the tests in VSCode or elsewhere, set the following in the proper environment:
 
 ```
-export PG_URL="postgresql://postgres@127.0.0.1:5432/teams_test?sslmode=disable&timezone=UTC"
+export PG_URL="postgresql://postgres@127.0.0.1:5432/teams_test?sslmode=disable&password=docker"
```
Comment (srenatus) [inline components/teams-service/README.md:32]: The UTC was there for a reason, wasn't it? I believe some tests mysteriously broke without it?

File: components/teams-service/server/team.go [hunk 2]
```diff
@@ -14,18 +14,18 @@ import (
 	"github.com/chef/automate/components/teams-service/storage"
 )
 
-// Server is a V2 teams server
-type Server struct {
+// TeamServer is a teams server
+type TeamServer struct {
```
Comment (srenatus) [inline components/teams-service/server/team.go:18]: [nit] Does this really need exporting?

File: components/teams-service/server/team_test.go [hunk 3]
```diff
@@ -51,15 +52,15 @@ func TestTeamsGRPC(t *testing.T) {
 
 		// If ciMode, run in-memory AND PG
 		// else just run PG.
-		if os.Getenv("CI") == "true, *authz.SubjectPurgeServerMock" {
+		if os.Getenv("CI") == "true" {
```
Comment (srenatus) [inline components/teams-service/server/team_test.go:55]: 👏

File: components/teams-service/README.md (hunk 1 above)
Comment (bcmdarroch) [reply components/teams-service/README.md:32]: oops didn't mean the drop that

File: components/teams-service/server/team.go (hunk 2 above)
Comment (bcmdarroch) [reply components/teams-service/server/team.go:18]: Tried changing this and the thread ended up tugging on the whole team test setup 😅 So I think I'll turn that into a separate cleanup card

File: components/teams-service/README.md [hunk 4]
```diff
@@ -29,7 +29,7 @@ make test_with_db
 If you wish to run the tests in VSCode or elsewhere, set the following in the proper environment:
 
 ```
-export PG_URL="postgresql://postgres@127.0.0.1:5432/teams_test?sslmode=disable&timezone=UTC"
+export PG_URL="postgresql://postgres@127.0.0.1:5432/teams_test?sslmode=disable&timezone=UTC&password=docker"
```
Comment (msorens) [inline components/teams-service/README.md:32]: 🎉

File: components/teams-service/server/server.go [hunk 5]
```diff
@@ -12,7 +12,6 @@ import (
 	"google.golang.org/grpc/status"
 
 	grpc_v2 "github.com/chef/automate/api/interservice/teams/v2"
```
Comment (msorens) [inline components/teams-service/server/server.go:14]: We cannot use just `grpc`, but should at least do this:
```suggestion
	grpc_iam "github.com/chef/automate/api/interservice/teams/v2"
```
And how much work to expunge the 'v2' on the end there...?

File: components/teams-service/service/service.go [hunk 6]
```diff
@@ -46,12 +46,7 @@ func NewPostgresService(l logger.Logger, connFactory *secureconn.Factory, migrat
 	authzSubjectClient authz.SubjectPurgeClient, authzV2PoliciesClient authz_v2.PoliciesClient,
 	authzV2AuthorizationClient authz_v2.AuthorizationClient) (*Service, error) {
```
Comment (msorens) [inline components/teams-service/service/service.go:47]:
```suggestion
	authzSubjectClient authz.SubjectPurgeClient, authzPoliciesClient authz_v2.PoliciesClient,
	authzAuthorizationClient authz_v2.AuthorizationClient) (*Service, error) {
```

File: components/teams-service/storage/conformance/conformance_test.go [hunk 7]
```diff
@@ -97,7 +97,7 @@ func TestStorage(t *testing.T) {
 
 		authzV2AuthorizationClient := authz_v2.NewAuthorizationClient(authzConn)
```
Comment (msorens) [inline components/teams-service/storage/conformance/conformance_test.go:98]:
```suggestion
		authzAuthorizationClient := authz_v2.NewAuthorizationClient(authzConn)
```
Also just a few lines above, s/mockV2Authz/mockAuthz/

File: components/teams-service/server/server.go (hunk 5 above)
Comment (bcmdarroch) [reply components/teams-service/server/server.go:14]: I was going to leave that for this card: https://github.com/chef/automate/issues/3481

File: components/teams-service/server/team.go (hunk 2 above)
Comment (srenatus) [reply components/teams-service/server/team.go:18]: It's fine. I thought it might be low-hanging. :shrug:

**Pull request merged**